test/e2e/router_status_test.go

```diff
@@ -0,0 +1,159 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TestDeleteIngressControllerShouldClearRouteStatus deletes an Ingress Controller and
+// makes sure a previously admitted route has its status cleared since.
+func TestDeleteIngressControllerShouldClearRouteStatus(t *testing.T) {
+	// Create an ingress controller that can admit our route.
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "ic-delete-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newLoadBalancerController(icName, domain)
```

Comment:
Better to use `newPrivateController` if you don't need a load balancer.  Provisioning an LB requires more time and money.  
```suggestion
	ic := newPrivateController(icName, domain)
```

Reply:
Agreed.

Done.